Upstream-Status: Backport

Signed-off-by: Maksim Sisov <msisov@igalia.com>
---
From fd42d45a0656d553b5fff5ede05553457596fd52 Mon Sep 17 00:00:00 2001
From: Nick Diego Yamane <nickdiego@igalia.com>
Date: Sat, 16 Mar 2019 18:17:33 +0000
Subject: [PATCH 08/27] ozone/wayland: Fix cursor regression when running in
 Weston

After crrev.com/c/1503374 regressions have been observed when
running chromium on Weston compositor. This cursor basically
remains with its default shape regardless the view is under
it (link, text, etc).

This CL fixes it by adding back the wl_surface::damage() call
before committing the cursor surface.

Bug: 942732
Change-Id: Ic96d6ee209424838f09e29a5a9eedff0e1591c83
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/1526465
Reviewed-by: Maksim Sisov <msisov@igalia.com>
Commit-Queue: Nick Yamane <nickdiego@igalia.com>
Cr-Commit-Position: refs/heads/master@{#641449}
---
 ui/ozone/platform/wayland/wayland_cursor.cc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/ui/ozone/platform/wayland/wayland_cursor.cc b/ui/ozone/platform/wayland/wayland_cursor.cc
index db2d5913550a..8c1f4bf5d9f4 100644
--- a/ui/ozone/platform/wayland/wayland_cursor.cc
+++ b/ui/ozone/platform/wayland/wayland_cursor.cc
@@ -69,6 +69,8 @@ void WaylandCursor::UpdateBitmap(const std::vector<SkBitmap>& cursor_image,
 
   wl_pointer_set_cursor(input_pointer_, serial, pointer_surface_.get(),
                         hotspot.x(), hotspot.y());
+  wl_surface_damage(pointer_surface_.get(), 0, 0, image_size.width(),
+                    image_size.height());
   wl_surface_attach(pointer_surface_.get(), buffer.get(), 0, 0);
   wl_surface_commit(pointer_surface_.get());
 
-- 
2.20.1

